src/test/java/uk/ac/ebi/atlas/experimentpage/markergenes/MarkerGeneTest.java

```diff
@@ -0,0 +1,82 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class MarkerGeneTest {
+    @Mock
+    private CellTypeSearchDao cellTypeSearchDaoMock;
+    @Mock
+    private MarkerGenesDao markerGenesDaoMock;
+
+    private MarkerGeneService subject;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        when(markerGenesDaoMock.getCellTypeMarkerGenes("E-EHCA-2", ImmutableSet.of("T cell", "B cell")))
+                .thenReturn(mockTestData());
+        subject = new MarkerGeneService(markerGenesDaoMock, cellTypeSearchDaoMock);
+    }
+
+    @Test
+    @DisplayName("Fetch marker gene profile expression from ontology label cell types")
+    void getMarkerGeneProfileWhenOntologyLabelsHasCellTypes() {
+        when(cellTypeSearchDaoMock.getInferredCellTypeOntologyLabels("E-EHCA-2", "skin"))
+                .thenReturn(ImmutableSet.of("T cell", "B cell"));
+        assertThat(subject.getCellTypeMarkerGeneProfile("E-EHCA-2", "skin"))
+                .isNotEmpty();
+    }
+
+    @Test
+    @DisplayName("Fetch marker gene profile expression from authors label cell types")
+    void getMarkerGeneProfileWhenAuthorsLabelsHasCellTypes() {
+        when(cellTypeSearchDaoMock.getInferredCellTypeAuthorsLabels("E-EHCA-2", "skin"))
+                .thenReturn(ImmutableSet.of("T cell", "B cell"));
+        when(cellTypeSearchDaoMock.getInferredCellTypeOntologyLabels("E-EHCA-2", "skin"))
+                .thenReturn(ImmutableSet.of());
+        assertThat(subject.getCellTypeMarkerGeneProfile("E-EHCA-2", "skin"))
+                .isNotEmpty();
+    }
+
+    @Test()
+    @DisplayName("Throws exception if both ontology labels returns empty cell type values")
+    void throwsExceptionIfBothOntologyAndAuthorsLabelsDoesNotHaveCellTypes() {
+        when(cellTypeSearchDaoMock.getInferredCellTypeOntologyLabels("E-EHCA-2", "skin"))
+                .thenReturn(ImmutableSet.of());
+        when(cellTypeSearchDaoMock.getInferredCellTypeAuthorsLabels("E-EHCA-2", "skin"))
+                .thenReturn(ImmutableSet.of());
+        assertThatThrownBy(() -> {
+            subject.getCellTypeMarkerGeneProfile("E-EHCA-2", "skin");
+        }).isInstanceOf(BioentityNotFoundException.class)
+                .hasMessageContaining("OrganismOrOrganismPart: skin doesn't have annotations.");
+    }
+
+    private List<CellTypeMarkerGene> mockTestData() {
+        List<CellTypeMarkerGene> CellTypeValuesMock = new ArrayList<>();
+        CellTypeValuesMock.add(CellTypeMarkerGene.create("1", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0,
+                "T cell", 125.258405, 563.54442));
+        CellTypeValuesMock.add(CellTypeMarkerGene.create("2", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0,
+                "T cell", 125.258405, 563.54442));
+        return CellTypeValuesMock;
```

Comment:
Remember to use immutable collections when possible. This could be rewritten as:
```java
private List<CellTypeMarkerGene> mockTestData() {
    return ImmutableList.of(
            CellTypeMarkerGene.create("1", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0, "T cell", 125.258405, 563.54442),
            CellTypeMarkerGene.create("2", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0, "T cell", 125.258405, 563.54442));
}
```

In fact, since the function is only called once, I would even inline it above as:
```java
when(
        markerGenesDaoMock.getCellTypeMarkerGenes("E-EHCA-2", ImmutableSet.of("T cell", "B cell")))
        .thenReturn(
                ImmutableList.of(
                        CellTypeMarkerGene.create("1", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0, "T cell", 125.258405, 563.54442),
                        CellTypeMarkerGene.create("2", "inferred cell type", "CD4-positive, alpha-beta T cell", 0.0, "T cell", 125.258405, 563.54442)));
```

In this way you remove one unncessary level of indirection and all the code needed to understand the mock stubbing is in the same place, no need to scroll up or down.

Reply:
Fixed it.